libcalico-go/lib/backend/k8s/k8s_test.go

```diff
@@ -1226,6 +1226,188 @@ var _ = testutils.E2eDatastoreDescribe("Test Syncer API for Kubernetes backend",
+			Ingress: []apiv3.Rule{
+				{
+					Action: "Allow",
+				},
+			},
+			Egress: []apiv3.Rule{
+				{
+					Action: "Allow",
+				},
+			},
+		}
+		stagedCalicoDisallowPolicyModelSpec := apiv3.StagedGlobalNetworkPolicySpec{
+			StagedAction: apiv3.StagedActionSet,
+			Order:        &zeroOrder,
+			Ingress: []apiv3.Rule{
+				{
+					Action: "Deny",
+				},
+			},
+			Egress: []apiv3.Rule{
+				{
+					Action: "Deny",
+				},
+			},
+		}
+
+		gnpClient := c.GetResourceClientFromResourceKind(apiv3.KindStagedGlobalNetworkPolicy)
+		kvp1Name := "my-test-sgnp"
+		kvp1KeyV1 := model.PolicyKey{Name: kvp1Name, Tier: "default"}
+		kvp1a := &model.KVPair{
+			Key: model.ResourceKey{Name: kvp1Name, Kind: apiv3.KindStagedGlobalNetworkPolicy},
+			Value: &apiv3.StagedGlobalNetworkPolicy{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       apiv3.KindStagedGlobalNetworkPolicy,
+					APIVersion: apiv3.GroupVersionCurrent,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: kvp1Name,
+				},
+				Spec: stagedCalicoAllowPolicyModelSpec,
+			},
+		}
+
+		kvp1b := &model.KVPair{
+			Key: model.ResourceKey{Name: kvp1Name, Kind: apiv3.KindStagedGlobalNetworkPolicy},
+			Value: &apiv3.StagedGlobalNetworkPolicy{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       apiv3.KindStagedGlobalNetworkPolicy,
+					APIVersion: apiv3.GroupVersionCurrent,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: kvp1Name,
+				},
+				Spec: stagedCalicoDisallowPolicyModelSpec,
+			},
+		}
+
+		kvp2Name := "my-test-sgnp2"
+		kvp2KeyV1 := model.PolicyKey{Name: kvp2Name, Tier: "default"}
+		kvp2a := &model.KVPair{
+			Key: model.ResourceKey{Name: kvp2Name, Kind: apiv3.KindStagedGlobalNetworkPolicy},
+			Value: &apiv3.StagedGlobalNetworkPolicy{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       apiv3.KindStagedGlobalNetworkPolicy,
+					APIVersion: apiv3.GroupVersionCurrent,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: kvp2Name,
+				},
+				Spec: stagedCalicoAllowPolicyModelSpec,
+			},
+		}
+
+		kvp2b := &model.KVPair{
+			Key: model.ResourceKey{Name: kvp2Name, Kind: apiv3.KindStagedGlobalNetworkPolicy},
+			Value: &apiv3.StagedGlobalNetworkPolicy{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       apiv3.KindGlobalNetworkPolicy,
+					APIVersion: apiv3.GroupVersionCurrent,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: kvp2Name,
+				},
+				Spec: stagedCalicoDisallowPolicyModelSpec,
+			},
+		}
+
+		// Check our syncer has the correct GNP entries for the two
+		// System Network Protocols that this test manipulates.  Neither
+		// have been created yet.
+		By("Checking cache does not have Staged Global Network Policy entries", func() {
+			Eventually(cb.GetSyncerValuePresentFunc(kvp1KeyV1)).Should(BeFalse())
+			Eventually(cb.GetSyncerValuePresentFunc(kvp2KeyV1)).Should(BeFalse())
+		})
+
+		By("Creating a Staged Global Network Policy", func() {
+			var err error
+			kvpRes, err = gnpClient.Create(ctx, kvp1a)
+			Expect(err).NotTo(HaveOccurred())
+		})
+		/*
```

Comment:
commented code? Can we remove or uncomment?

Reply:
I'll address this in the next PR: https://github.com/mazdakn/calico/pull/2